packages/DatePicker/src/DatePicker.js

```diff
@@ -201,19 +103,19 @@ function DatePicker(props) {
   }
 
   function handleSelect(selectedDate) {
-    setHasParsingError(false);
     hideCalendar();
     handleChange(selectedDate);
   }
 
-  const hasInputError = extendedInputProps && extendedInputProps.hasError;
+  function handlePossibleDateChange(newPossibleDate) {
+    if (newPossibleDate.isSame(possibleDate, "year") && newPossibleDate.isSame(possibleDate, "month")) return;
+
+    setPossibleDate(newPossibleDate);
+  }
 
-  const hasErrorValue = hasError || hasParsingError || hasInputError;
+  const hasInputError = extendedInputProps && extendedInputProps.hasError;
```

Comment:
Might be nice if extractChildrenProps returned `{}` instead of undefined, so we could skip the defensive check here.